ci/requirements-3.6-dev-pyspark-spark.yml

```diff
@@ -0,0 +1,59 @@
+channels:
+  - conda-forge
+dependencies:
+  - black=19.10b0
+  - click
+  - clickhouse-cityhash
+  - clickhouse-driver>=0.1.3
+  - clickhouse-sqlalchemy
+  - cmake
+  - flake8
+  - geoalchemy2
+  - geopandas
+  - google-cloud-bigquery>=1.0.0
+  - graphviz
+  - impyla>=0.15.0
+  - jinja2
+  - libiconv  # see https://github.com/jupyter/repo2docker/issues/758
+  - lz4
+  - multipledispatch>=0.6.0
+  - mypy
+  - numpy>=1.15
+  - openjdk=8
+  - pandas>=0.25.3
+  - pip=19.3.1
+  - plumbum
+  - pre-commit
+  - psycopg2
+  - pyarrow>=0.13, <=0.14.1
```

Comment:
These new `requirements-3.x-dev-pyspark-spark.yml` files are the same as `requirements-3.x-dev.yml` except `pymapd` is disabled and `<=0.14.1` is added to the `pyarrow` version pin

Reply:
Does it make sense to remove all the non spark dependency here? e.g., pymysql, impyla, bigquery, clickhouse, etc? I think

Reply:
I think to run Pyspark CI you actually only need very minimal dependencies, i.e., pyspark, pyarrow, pandas and numpy

Reply:
That makes sense to me, I'll do that

Reply:
Hopefully reduces the Conda solve time too, it's highest in the PySpark/Spark CI jobs

Reply:
can you add a comment / issue link about this (do we have an open ibis issue for this?)

Reply:
Do you mean a comment/issue about the solve time? I think in general the solve time isn't really an issue now after [this commit](https://github.com/ibis-project/ibis/commit/3d9756cee1dc3e0c622891dd6314c2f0abecc382)—I just noticed that Conda takes a while to solve in the new PySpark/Spark CI jobs I'm adding here. Removing the extra dependencies would speed it up though and make the CI finish a bit faster

Reply:
Oh I just saw you commented this also on the other requirements file, do you mean create an issue about the pyarrow version (increasing it, or changing supported versions)?